include/zephyr/llext/llext.h

```diff
@@ -177,6 +179,48 @@ int llext_load(struct llext_loader *loader, const char *name, struct llext **ext
  */
 int llext_unload(struct llext **ext);
 
+/** @brief Entry point function signature for an extension. */
+typedef void (*llext_entry_fn_t)(void *user_data);
+
+/**
+ * @brief Setup and execute an extension.
+ *
+ * Call the extension's own initialization functions, then an additional
+ * entry point in the same context, if desired.
+ *
+ * This function matches the @ref k_thread_entry_t signature, so it
+ * can either be run directly or as a new user or kernel thread via
+ * @ref k_thread_create.
+ *
+ * @param[in] ext Extension to initialize and execute. Passed as `p1` in
+ *                @ref k_thread_create.
+ * @param[in] entry_fn Main entry point of the thread after performing
+ *                     extension initialization. Passed as `p2` in
+ *                     @ref k_thread_create. May be `NULL`.
+ * @param[in] user_data Argument passed to @a entry_fn. Passed as `p3`
+ *                      in @ref k_thread_create.
+ */
+void llext_bootstrap(struct llext *ext, llext_entry_fn_t entry_fn, void *user_data);
```

Comment:
No. Since `llext_bootstrap` is the "launch vector" for the init functions, it definitely must run in user mode if the app writer wants. The elevated-privileges part is in `llext_get_init_array`.